app/services/course/assessment/programming_evaluation_service.rb

```diff
@@ -107,7 +108,7 @@ def execute
   def create_evaluation
     Course::Assessment::ProgrammingEvaluation.create(
       course: @course, language: @language, package_path: @package, memory_limit: @memory_limit,
-      time_limit: @time_limit.to_i
+      time_limit: @time_limit
```

Comment:
It's enforced by the db and the validations already.

The evaluator passes whatever it gets to the `ulimit` command. Numbers seem to work, and so does the string `unlimited`.